pkg/cluster/internal/kubeadm/config.go

```diff
@@ -237,6 +239,7 @@ evictionHard:
   nodefs.available: "0%"
   nodefs.inodesFree: "0%"
   imagefs.available: "0%"
+resolvConf: /kind/resolv.conf
```

Comment:
I really don't like the idea that containerd and debugging tools will have a different resolv conf than kubelet.

Reply:
as you said 
> we could selectively merge the resolv.conf

Reply:
After considering it, I think the current practice is relatively simple. If we do resolv.conf merge, we need a lot of testing. Why don't we put it later? :thinking: 

Keep this practice for a while, then change it after the verification is passed.

I hope we can move forward. :heart:

Reply:
this means other components have an inconsistent resolv conf and we're not actually solving the issue. if we're going to use hostnames we need kubelet and everything else to have the same lookup.

Reply:
technically having a different resolv.conf doesn't mean that have a different lookup, just that the pods will not be able to resolv the nodes names, the other requests will use the same dns server, however I agree that reality is always different and we can be broken because docker changes on the DNS behaviour, that will not be surprising 😅 

> Docker daemon runs an embedded DNS server which provides DNS resolution among containers connected to the same user-defined network, so that these containers can resolve container names to IP addresses. If the embedded DNS server is unable to resolve the request, it will be forwarded to any external DNS servers configured for the container. To facilitate this when the container is created, only the embedded DNS server reachable at 127.0.0.11 will be listed in the container’s resolv.conf file. 

I agree with tao here, I think that we should left until last the DNS exercise

Reply:
+1 on merging the resolv.confs if feasble.